cdbot/cogs/quoting.py

```diff
@@ -0,0 +1,382 @@
+from math import ceil
+from typing import Callable, List, Union
+
+from discord import Colour, Embed, HTTPException, Member, Message, NotFound, RawReactionActionEvent, TextChannel, User
+from discord.ext import commands
+from discord.ext.commands import ArgumentParsingError, Bot, CheckFailure, Cog, UserConverter
+from motor.motor_asyncio import AsyncIOMotorClient
+
+from cdbot.constants import (
+    CYBERDISC_ICON_URL, LOGGING_CHANNEL_ID, MongoDB, QUOTES_CHANNEL_ID, QUOTES_DELETION_QUOTA, QUOTE_CZAR_ID, SERVER_ID
+)
+
+
+def is_quote_czar() -> Callable:
+    async def predicate(ctx: commands.Context) -> bool:
+        czar_role = ctx.guild.get_role(QUOTE_CZAR_ID)
+        return czar_role in ctx.author.roles or ctx.author.guild_permissions.administrator
+    return commands.check(predicate)
+
+
+class FormerUser(UserConverter):
+    async def convert(self, ctx, argument):
+        try:
+            return await ctx.bot.fetch_user(argument)
+        except (NotFound, HTTPException):
+            return await super().convert(ctx, argument)
+
+
+class QuoteCog(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_ready(self):
+        self.guild = self.bot.get_guild(SERVER_ID)
+        self.quote_channel = self.bot.get_channel(QUOTES_CHANNEL_ID)
+        self.logs_channel = self.bot.get_channel(LOGGING_CHANNEL_ID)
+        self.database = AsyncIOMotorClient(
+            host=MongoDB.MONGOHOST,
+            port=MongoDB.MONGOPORT,
+            username=MongoDB.MONGOUSER,
+            password=MongoDB.MONGOPASSWORD,
+        )[MongoDB.MONGODATABASE]["quotes"]
+
+    @Cog.listener()
+    async def on_command_error(self, ctx: commands.Context, error):
+        await ctx.send(f"{type(error)}: {error}")
+
+    @Cog.listener()
+    async def on_raw_reaction_add(self, raw_reaction: RawReactionActionEvent):
+        thumbs_down = "\N{THUMBS DOWN SIGN}"
+        if str(raw_reaction.emoji) == thumbs_down and raw_reaction.channel_id == QUOTES_CHANNEL_ID:
+            message = await self.quote_channel.fetch_message(raw_reaction.message_id)
+            reaction = [react for react in message.reactions if str(react.emoji) == thumbs_down][0]
+            if reaction.count > QUOTES_DELETION_QUOTA:
+                await self.database.delete_one({"_id": message.id})
```

Comment:
Consider moving this part to a function? I guess if it's not repeated it's not a massive deal but this seems like  it's just `delete_quote()`, and it'd make it easier to see what's happening here at a glance.

Reply:
This is true, it can be done but I think simple annotations will help.

Reply:
Yeah I'll add some comments for this in the code, no real need for a function here as you've said.